### ISSUE TITLE
Date and Time Formats were incorrectly applied when displaying dates in Korean (or other) languages.

### ISSUE THREAD

**Issue by fofwisdom** [mastodon/mastodon#26836]
### Steps to reproduce the problem

1. Change the language to Korean
2. open the any posts
example - https://muri.network/deck/@TEA_FOP/111000303520826018
<img width="557" alt="스크린샷 2023-09-07 15 39 19" src="https://github.com/mastodon/mastodon/assets/205521/3727e5b3-abb9-48ab-ab71-d0b5c3f1fd05">

### Expected behaviour

two digit (2023년 09월 07일 or 2023-09-07) (%Y-%m-%d) 

### Actual behaviour

mixed one digit & two digit (2023년 9월 07일 or 2023-9-07) (%Y-%f-%d)

### Detailed description

Date and Time Formats - https://www.ibm.com/docs/en/cmofz/10.5.0?topic=reference-date-time-formats
Korea translation - https://crowdin.com/translate/mastodon/all/en-ko?filter=basic&value=0#q=%25Y

The Korean translation only calls %f in two cases, and does not write %f and %d together.

### Mastodon instance

mastodon.social / muri.network

### Mastodon version

v4.2.0-nightly.2023-09-06 / v4.2.0-beta3

### Browser name and version

Microsoft Edge 116.0.1938.69 / Chrome 116.0.5845.164

### Operating system

macOS 13.5.1 / Android 13

**Comment by ClearlyClaire** [comment]
> Korea translation - https://crowdin.com/translate/mastodon/all/en-ko?filter=basic&value=0#q=%25Y

These translation strings are for server-side-rendered content, which is not the case in your screenshot.

As far as I can tell, the code responsible for your screenshot is:
https://github.com/mastodon/mastodon/blob/8a9d7aeb1effff661373f18157a87eb607a0000f/app/javascript/mastodon/features/status/components/detailed_status.jsx#L326

It appears it sets `numeric` for year, `short` for month (which in English locale is not numeric), and `2-digit` for day. So I suppose the issue is the `short` style for months in Korean is numeric?

**Comment by fofwisdom** [comment]
Hi ClearlyClaire,

Thanks for the feedback.

If we write X year Y month Z day or month name, we need to hide the zeros in the number of days. Only Mastodon uses two digits, including zeros, in its detailed date format.

Unless our translation project can control the date formatting of posts, we'll have to leave the zeros out. It's a minor change, so even if it's an afterthought, it seems like something that should be fixed at someday.

Mastodon
![image](https://github.com/mastodon/mastodon/assets/205521/d48ad79b-9fe3-43ec-9a69-07cd0253a70d)

Wikipedia
![image](https://github.com/mastodon/mastodon/assets/205521/f786114d-6c7e-4875-9ad4-8a0cb48081b2)

Google Play
![image](https://github.com/mastodon/mastodon/assets/205521/8c7189d8-4bdf-48d5-9434-044fa18c0f53)

Microsoft Windows (Germany)
![image](https://github.com/mastodon/mastodon/assets/205521/90ca3129-7cac-48c6-9b47-c8d08d7b8f16)

Microsoft Windows (English Germany)
![image](https://github.com/mastodon/mastodon/assets/205521/2b34a766-a05d-4305-8e80-5e9eeda203cc)

Microsoft Windows (English USA)
![image](https://github.com/mastodon/mastodon/assets/205521/9be0d3b7-9bef-4fb7-aa9a-aa16ce87d71f)

So many thanks.

**Comment by fofwisdom** [comment]
and Inside the dashboard in Admin, I found that it was displaying correctly.

![image](https://github.com/mastodon/mastodon/assets/205521/af3b4712-1abb-43c7-bc3a-d8b7aeabe955)

**Comment by ClearlyClaire** [comment]
Yeah, you're right, using `2-digits` for days is a weird choice regardless of the locale!
